agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -0,0 +1,609 @@
+
+	// map to transform credentialspec values, key is a input credentialspec
+	// Examples:
+	// * key := credentialspec:file://credentialspec.json, value := credentialspec=file://credentialspec.json
+	// * key := credentialspec:s3ARN, value := credentialspec=file://CredentialSpecResourceLocation/s3_taskARN_fileName.json
+	// * key := credentialspec:ssmARN, value := credentialspec=file://CredentialSpecResourceLocation/ssm_taskARN_param.json
+	CredSpecMap map[string]string
+
+	// lock is used for fields that are accessed and updated concurrently
+	lock sync.RWMutex
+}
+
+// NewCredentialSpecResource creates a new CredentialSpecResource object
+func NewCredentialSpecResource(taskARN, region string,
+	credentialSpecs map[string][]*apicontainer.Container,
+	executionCredentialsID string,
+	credentialsManager credentials.Manager,
+	ssmClientCreator ssmfactory.SSMClientCreator,
+	s3ClientCreator s3factory.S3ClientCreator) (*CredentialSpecResource, error) {
+
+	s := &CredentialSpecResource{
+		taskARN:                 taskARN,
+		region:                  region,
+		requiredCredentialSpecs: credentialSpecs,
+		credentialsManager:      credentialsManager,
+		executionCredentialsID:  executionCredentialsID,
+		ssmClientCreator:        ssmClientCreator,
+		s3ClientCreator:         s3ClientCreator,
+		CredSpecMap:             make(map[string]string),
+	}
+
+	err := s.setCredentialSpecResourceLocation()
+	if err != nil {
+		return nil, err
+	}
+
+	s.initStatusToTransition()
+	return s, nil
+}
+
+func (cs *CredentialSpecResource) initStatusToTransition() {
+	resourceStatusToTransitionFunction := map[resourcestatus.ResourceStatus]func() error{
+		resourcestatus.ResourceStatus(CredentialSpecCreated): cs.Create,
+	}
+	cs.resourceStatusToTransitionFunction = resourceStatusToTransitionFunction
+}
+
+func (cs *CredentialSpecResource) Initialize(resourceFields *taskresource.ResourceFields,
+	taskKnownStatus status.TaskStatus,
+	taskDesiredStatus status.TaskStatus) {
+	cs.initStatusToTransition()
+	cs.credentialsManager = resourceFields.CredentialsManager
+	cs.ssmClientCreator = resourceFields.SSMClientCreator
+	cs.s3ClientCreator = resourceFields.S3ClientCreator
+
+	// if task hasn't turn to 'created' status, and it's desire status is 'running'
+	// the resource status needs to be reset to 'NONE' status so the cs value
+	// will be retrieved again
+	if taskKnownStatus < status.TaskCreated &&
+		taskDesiredStatus <= status.TaskRunning {
+		cs.SetKnownStatus(resourcestatus.ResourceStatusNone)
+	}
+}
+
+// GetTerminalReason returns an error string to propagate up through to task
+// state change messages
+func (cs *CredentialSpecResource) GetTerminalReason() string {
+	return cs.terminalReason
+}
+
+func (cs *CredentialSpecResource) setTerminalReason(reason string) {
+	cs.terminalReasonOnce.Do(func() {
+		seelog.Infof("credentialspec resource: setting terminal reason for credentialspec resource in task: [%s]", cs.taskARN)
```

Comment:
not sure if this will cause lots of unnecessary logs in agent log, maybe log as debug?

Reply:
Done, thanks!